### PR TITLE
Add the initial set of commands

### PR DESCRIPTION
**Pull request by mykola-mokhnach**
Part of https://github.com/appium/appium/issues/12631

@KazuCocoa  FYI

### CODE REVIEW

File: lib/tools/app-commands.js [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+import { fixOutputToArray } from '../helpers';
+
+const appCommands = {};
+
+/**
+ * Lists the targets installed applications and their metadata.
+ * Example output:
+ * {"bundle_id": "com.apple.test.IntegrationTests_1-Runner", "name": "IntegrationTests_1-Runner", "install_type": "user", "architectures": ["x86_64", "i386"], "process_state": "Unknown", "debuggable": false}
+ * {"bundle_id": "com.apple.mobilesafari", "name": "MobileSafari", "install_type": "system", "architectures": ["x86_64"], "process_state": "Unknown", "debuggable": false}
+ *
```
Comment (KazuCocoa) [inline lib/tools/app-commands.js]: On a real device, the result is like:

```
{"bundle_id": "com.cookpad", "name": "\u30af\u30c3\u30af\u30d1\u30c3\u30c9", "install_type": "user", "architectures": null, "process_state": "Unknown", "debuggable": false}
```

File: lib/tools/app-commands.js [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    failIfRunning = false,
+    wait = false,
+  } = opts;
+  log.debug(`Launching '${bundleId}' on the device '${this.udid}'`);
+  const args = [];
+  if (!failIfRunning) {
+    args.push('--foreground-if-running');
+  }
+  if (!wait) {
+    await this.exec(['launch', bundleId, ...args]);
```
Comment (KazuCocoa) [inline lib/tools/app-commands.js]: On a real device, we can get below error

```
$ idb launch --udid 242bfcc998fd156df0bd0ad1dde8ab8e0a032114 com.cookpad
Application Bundle Name: クックパッド | ID: com.cookpad | Install Type user | Container (null) cannot be launched as it's not signed with a development identity
$ echo $?
1
```

File: lib/tools/accessibility-commands.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+/**
+ * Describes the Accessibility tree on the device under test
+ *
+ * @returns {Array<object>} JSON formatted list of all the elements currently on screen,
+ * including their bounds and accessibility information
+ */
+accessibilityCommands.describeAll = async function describeAll () {
+  log.debug(`Describing the UI on the device '${this.udid}'`);
+  return JSON.parse(await this.exec(['ui', 'describe-all']));
```
Comment (KazuCocoa) [inline lib/tools/accessibility-commands.js]: below error can happen.

```
$ idb ui describe-all --udid A82163AC-8AD8-4ABA-AC7E-F2A3F95593ED
Could not enable accessibility for bridge '<SimulatorBridge: 0x7fa3cd701d70>'
$ echo $?
1
```

mm, I got below in another time...
```
$ idb ui describe-all --udid 322FC4CB-5D1A-4D53-9D28-8BC0DA696310
Connection was closed
```

File: lib/tools/app-commands.js (hunk 2 above)
Comment (mykola-mokhnach) [reply lib/tools/app-commands.js]: this is fine. Both return non-zero code and will trigger an exception from `exec` call

File: lib/tools/accessibility-commands.js (hunk 3 above)
Comment (mykola-mokhnach) [reply lib/tools/accessibility-commands.js]: same as above

File: lib/helpers.js [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+ */
+function fixOutputToArray (output) {
+  if (!_.trim(output)) {
+    return [];
+  }
+
+  return output.split('\n')
+    .map((x) => x.trim())
+    .filter(Boolean)
+    .filter((x) => {
```
[the remark this answers is not in the document]
Comment (mykola-mokhnach) [reply lib/helpers.js]: using reduce now

File: lib/helpers.js [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+async function getPids (pattern, opts = {}) {
+  const {
+    multi = true,
+    ignoreCase = true,
+  } = opts;
+  const args = [`-${ignoreCase ? 'i' : ''}f${multi ? '' : 'n'}`, pattern];
+  try {
+    const {stdout} = await exec('pgrep', args);
+    const result = stdout.split('\n')
+      .filter((x) => parseInt(x, 10))
```
[the remark this answers is not in the document]
Comment (mykola-mokhnach) [reply lib/helpers.js]: fixed

**Pull request merged**